### DISCUSSION TITLE
Device Unavailable on HA but always available on Tapo App

### DISCUSSION THREAD

**Discussion by ciquaal** [petretiandrea/home-assistant-tapo-p100#553] [General]
Afternoon all,

I didn't want to label this one as a bug as I am unsure on the root cause..

I have a Tapo hub, Tapo thermostat and some P100's installed to monitor a greenhouse.
I keep having a fair few of my devices drop one at a time, or even all of them at once from HA but they do not disconnect from my Tapo App.

All devices are connected to the same Wi-Fi and there is full Wi-Fi coverage of the greenhouse and home.
Has anyone else experienced such things and potentially have a solution?

**Comment by Bodfrey** [comment]
I'm having lots of problems here - see this #648
